model-optimizer/extensions/middle/L2NormToNorm.py

```diff
@@ -82,6 +82,27 @@ def replace_pattern(self, graph: Graph, match: dict):
             log.debug('The value of the "maximum_y_data" is not defined or is not constant')
             return
 
+        # We need to check axes which performed reduction because IE supports only 2D, 3D, 4D inputs and
+        # reduction only along spatial dimensions.
+        input_rank = len(match['sum'].in_port(0).data.get_shape())
+        if input_rank not in [2, 3, 4]:
+            log.debug('IE supports L2 normalization only for 2D, 3D and 4D tensors, skip fusing transformation.')
+            return
+
+        axes = match['sum'].in_port(1).data.get_value()
+        axes = int64_array(axes)
+        if axes.shape == ():
+            axes = int64_array([axes])
+        axes.sort()
+
```

Comment:
I think that `channel_shared` should be set up based `axes` as well as `across_spatial`. Now you ignore channel dimension.